crates/iota-names-indexer/src/worker.rs

```diff
@@ -120,6 +126,32 @@ impl Worker for IotaNamesWorker {
             checkpoint.checkpoint_summary.sequence_number
         );
 
+        let config_type = StructTag::from_str(&format!(
+            "{}::iota_names::BalanceKey<iota::iota::IOTA>",
+            self.config.package_address,
+        ))?;
+        let layout = MoveTypeLayout::Struct(Box::new(MoveStructLayout {
+            type_: config_type.clone(),
+            fields: vec![MoveFieldLayout::new(
+                Identifier::from_str("dummy_field")?,
+                MoveTypeLayout::Bool,
+            )],
+        }));
+        let balance_object_id = iota_types::dynamic_field::derive_dynamic_field_id(
+            self.config.object_id,
+            &TypeTag::Struct(Box::new(config_type)),
+            &IotaJsonValue::new(serde_json::json!({ "dummy_field": false }))?
+                .to_bcs_bytes(&layout)?,
+        )?;
```

Comment:
Any way to not construct these 3 seemingly constant things for every single checkpoint?

Reply:
Yes, but it might get messy. TBH I hate this code. I will be looking for a better way tomorrow

Reply:
I will open an issue then

Reply:
https://github.com/iotaledger/iota/issues/7163